app/src/ui/diff/text-diff.tsx

```diff
@@ -1004,6 +1007,11 @@ export class TextDiff extends React.Component<ITextDiffProps, ITextDiffState> {
     if (batchedOps.length > 0) {
       cm.operation(() => batchedOps.forEach(x => x()))
     }
+
+    // We have to set the size of the gutter, and because it might have changed we should refresh to fit the editor.
+    const gutterElement = document.getElementsByClassName('diff-gutter')[0]
+    gutterElement.setAttribute('style', `width: ${diffSize * 2}px;`)
+    cm.refresh()
```

Comment:
Maybe there is some other place where you can apply the style changes and you don't need to call `refresh()`? Maybe in the `render` method? 🤔

Reply:
Ohhh ok, that makes sense!

Re obtaining the `diff-gutter` element: can you use `cm.getGutterElement()` instead? That gives you the parent of the element your code is getting now. If setting the `width` to that doesn't work, you can at least use that element instead of `document` to call `getElementsByClassName('diff-gutter')` which I'd hope would be more efficient as it should narrow the search.

I wonder if there is a way to encapsulate this logic inside `CodeMirrorHost`, so that the `refresh` is not needed (for example, adding another prop to `CodeMirrorHost` with the gutter width and apply it to the gutter element in the `updateDoc` method?).

Reply:
So I swapped it to use the getGutterElement and that works fine. I've tried the method you mentioned for doing it inside of the updateDoc method, but even if a different size is passed to updateDoc and it's set on the correct element the size of the editor never changes. I might be doing it incorrectly or not understanding, but I didn't see anything inside the updateDoc method that would refresh code mirror to resize itself.